src/chips/add_carry_v1.rs

```diff
@@ -0,0 +1,162 @@
+        instance: Column<Instance>,
+    ) -> AddCarryConfig {
+        let col_a = advice[0];
+        let col_b = advice[1];
+        let col_c = advice[2];
+        let add_carry_selector = selector;
+
+        // Enable equality on the advice and instance column to enable permutation check
+        meta.enable_equality(col_a);
+        meta.enable_equality(col_b);
+        meta.enable_equality(col_c);
+        meta.enable_equality(instance);
+
+        // enforce dummy hash function by creating a custom gate
+        meta.create_gate("accumulate constraint", |meta| {
+            let s = meta.query_selector(add_carry_selector);
+            let prev_b = meta.query_advice(col_b, Rotation::prev());
+            let prev_c = meta.query_advice(col_c, Rotation::prev());
+            let a = meta.query_advice(col_a, Rotation::cur());
+            let b = meta.query_advice(col_b, Rotation::cur());
+            let c = meta.query_advice(col_c, Rotation::cur());
+
+            // Previous accumulator amount + new value from a_cell
+            // using binary expression (x_n-4 * 2^16) + (x_n-3 * 2^8) + ... + (x_n * 2)
+            vec![
+                s * ((a + (prev_b * Expression::Constant(Fp::from(1 << 16))) + prev_c)
+                    - ((b * Expression::Constant(Fp::from(1 << 16))) + c)),
+            ]
+        });
+
+        AddCarryConfig {
+            advice: [col_a, col_b, col_c],
+            instance,
+            selector: add_carry_selector,
+        }
+    }
+
+    // Initial accumulator values from instance for expreiment
+    pub fn assign_first_row(
+        &self,
+        mut layouter: impl Layouter<Fp>,
+    ) -> Result<(AssignedCell<Fp, Fp>, AssignedCell<Fp, Fp>), Error> {
+        layouter.assign_region(
+            || "first row",
+            |mut region| {
+                let b_cell = region.assign_advice_from_instance(
+                    || "first acc[1]",
+                    self.config.instance,
+                    0,
+                    self.config.advice[1],
+                    0,
+                )?;
+
+                let c_cell = region.assign_advice_from_instance(
+                    || "first acc[2]",
+                    self.config.instance,
+                    1,
+                    self.config.advice[2],
+                    0,
+                )?;
+
+                Ok((b_cell, c_cell))
+            },
+        )
+    }
+
+    pub fn assign_advice_row(
+        &self,
+        mut layouter: impl Layouter<Fp>,
+        a: Value<Fp>,
+        prev_b: AssignedCell<Fp, Fp>,
+        prev_c: AssignedCell<Fp, Fp>,
+    ) -> Result<(AssignedCell<Fp, Fp>, AssignedCell<Fp, Fp>), Error> {
+        layouter.assign_region(
+            || "adivce row for accumulating",
+            |mut region| {
+                // enable hash selector
+                self.config.selector.enable(&mut region, 1)?;
+
+                let _ = prev_b.copy_advice(|| "prev_b", &mut region, self.config.advice[1], 0);
```

Comment:
I see that you are copying prev_b and prev_c from the previous region here. So basically the row gets duplicated across the two regions. Can we somehow avoid that? Maybe it can work with a single region.

Reply:
If we are dealing with multiple values on the `add_carry_v1` circuit, I think it's right to do it like the example you have given. 
also, we can avoid `copy_advice` by using two more inputs to the `assign_advice_row` method like this.
```Rust
pub fn assign_advice_row(
    &self,
    mut layouter: impl Layouter<Fp>,
    a: Value<Fp>,
    prev_b: Value<Fp>,
    prev_c: Value<Fp>,
    b: Value<Fp>,
    c: Value<Fp>,
```
and then `prev_b = b`, `prev_c = c` in somewhere circuit.

However, in this case, I opted for a simpler approach by adding a new value to the previous two columns, which were assigned values from instances at the beginning. Additionally, the `copy_advice` function has an equality check, which is very useful when used at other circuits or chips